src/fragments/start/getting-started/flutter/integrate.mdx

```diff
@@ -228,230 +202,370 @@ class _AddTodoFormState extends State<AddTodoForm> {
 
 ```bash
 flutter run
 ```
 
-Let’s next configure Amplify and give the app a reason to have a loading state at all.
+![Budget Tracker home screen](/images/lib/getting-started/flutter/budget-tracker-home.png)
 
 ## Configure Amplify
 
-To the top of the `_TodosPageState` class, add the following.
-
-```dart
-class _TodosPageState extends State<TodosPage> {
-
-  // loading ui state - initially set to a loading state
-  bool _isLoading = true;
-
-  // list of Todos - initially empty
-  List<Todo> _todos = [];
-```
-
-Update the `initState()` function in the `_TodosPageState` class.
-
-```dart
-@override
-void initState() {
-
-  // kick off app initialization
-  _initializeApp();
+Let's start by configuring Amplify when the application loads. At the top of the file, find the `_configureAmplify`
+method and replace it with the following code.
 
-  super.initState();
-}
-```
-
-Update the `_initializeApp()` function in the `_TodosPageState` class.
+```diff
++ import 'package:amplify_api/amplify_api.dart';
++ import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
++ import 'package:amplify_authenticator/amplify_authenticator.dart';
++ import 'package:amplify_flutter/amplify_flutter.dart';
+import 'package:flutter/material.dart';
+import 'package:go_router/go_router.dart';
 
-```dart
-Future<void> _initializeApp() async {
++ import 'amplifyconfiguration.dart';
+import 'models/ModelProvider.dart';
 
-  // configure Amplify
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
   await _configureAmplify();
+  runApp(const MyApp());
+}
 
-  // after configuring Amplify, update loading ui state to loaded state
-  setState(() {
-    _isLoading = false;
-  });
+Future<void> _configureAmplify() async {
+-  // To be filled in
++  try {
++    // Create the API plugin.
++    //
++    // If `ModelProvider.instance` is not available, try running 
++    // `amplify codegen models` from the root of your project.
++    final api = AmplifyAPI(modelProvider: ModelProvider.instance);
++
++    // Create the Auth plugin.
++    final auth = AmplifyAuthCognito();
++
++    // Add the plugins and configure Amplify for your app.
++    await Amplify.addPlugins([api, auth]);
++    await Amplify.configure(amplifyconfig);
++
++    safePrint('Successfully configured');
++  } on Exception catch (e) {
++    safePrint('Error configuring Amplify: $e');
++  }
 }
 ```
 
-Update the `_configureAmplify()` function in the `_TodosPageState` class.
+Here, we're adding the API and Authentication plugins to our app and configuring Amplify with the
+generated `amplifyconfiguration.dart` file.
 
-```dart
-Future<void> _configureAmplify() async {
-  try {
-
-    // amplify plugins
-    final _dataStorePlugin = AmplifyDataStore(modelProvider: ModelProvider.instance);
+There's one more step to complete the configuration of Auth and that is to wrap our application
+in the Amplify Authenticator, which will provide a pre-built authentication flow with no setup required.
```

Comment:
"which will provide a pre-built authentication flow with less than 5 lines of code"